global: better PEP-8 and PEP-257 compliance

The PEP-8 and PEP-257 compliance should be improved so that we can plug kwalitee checks for Python code in the Travis CI.

```console
claimstore/__init__.py:1:1: D100  Missing docstring in public module
claimstore/wsgi.py:1:1: D100  Missing docstring in public module
claimstore/wsgi.py:2:1: I201 Missing newline before sections or imports.
claimstore/modules/__init__.py:1:1: D100  Missing docstring in public module
claimstore/modules/claims/__init__.py:1:1: D100  Missing docstring in public module
claimstore/modules/claims/views.py:1:1: D100  Missing docstring in public module
claimstore/modules/claims/views.py:1:1: I101 Imported names are in the wrong order. Should be Blueprint, render_template
claimstore/modules/claims/views.py:2:1: I100 Imports statements are in the wrong order. from claimstore.core.json should be before from flask
claimstore/modules/claims/views.py:2:1: I201 Missing newline before sections or imports.
claimstore/modules/claims/views.py:12:1: D103  Missing docstring in public function
claimstore/modules/claims/views.py:17:1: D103  Missing docstring in public function
claimstore/modules/claims/views.py:18:80: E501 line too long (95 > 79 characters)
claimstore/modules/claims/views.py:22:1: D103  Missing docstring in public function
claimstore/modules/claims/views.py:27:1: D103  Missing docstring in public function
claimstore/core/__init__.py:1:1: D100  Missing docstring in public module
claimstore/app.py:1:1: D100  Missing docstring in public module
claimstore/app.py:1:1: I101 Imported names are in the wrong order. Should be Flask, jsonify, render_template
claimstore/app.py:3:1: I100 Imports statements are in the wrong order. from claimstore.core.exception should be before from flask.ext.sqlalchemy
claimstore/app.py:3:1: I201 Missing newline before sections or imports.
claimstore/app.py:10:1: D103  Missing docstring in public function
claimstore/app.py:16:1: D103  Missing docstring in public function
claimstore/core/decorators.py:1:1: D100  Missing docstring in public module
claimstore/core/decorators.py:2:1: I201 Missing newline before sections or imports.
claimstore/core/decorators.py:5:1: D103  Missing docstring in public function
claimstore/core/decorators.py:10:80: E501 line too long (90 > 79 characters)
claimstore/core/exception.py:1:1: D100  Missing docstring in public module
claimstore/core/exception.py:1:1: D101  Missing docstring in public class
claimstore/core/exception.py:4:1: D102  Missing docstring in public method
claimstore/core/exception.py:11:1: D102  Missing docstring in public method
claimstore/core/json.py:1:1: D100  Missing docstring in public module
claimstore/core/json.py:2:1: I100 Imports statements are in the wrong order. import json should be before import os
claimstore/core/json.py:3:1: I201 Missing newline before sections or imports.
claimstore/core/json.py:4:1: I100 Imports statements are in the wrong order. from flask should be before import jsonschema
claimstore/core/json.py:4:1: I201 Missing newline before sections or imports.
claimstore/core/json.py:7:1: D202  No blank lines allowed after function docstring (found 1)
claimstore/core/json.py:10:80: E501 line too long (89 > 79 characters)
claimstore/core/json.py:28:1: D202  No blank lines allowed after function docstring (found 1)
claimstore/modules/claims/models.py:1:1: D100  Missing docstring in public module
claimstore/modules/claims/models.py:2:1: I101 Imported names are in the wrong order. Should be JSONB, UUID
claimstore/modules/claims/models.py:2:1: I201 Missing newline before sections or imports.
claimstore/modules/claims/models.py:3:1: I100 Imports statements are in the wrong order. from claimstore.app should be before from sqlalchemy.dialects.postgresql
claimstore/modules/claims/models.py:3:1: I201 Missing newline before sections or imports.
claimstore/modules/claims/models.py:6:1: D101  Missing docstring in public class
claimstore/modules/claims/models.py:59:1: D102  Missing docstring in public method
claimstore/modules/claims/models.py:63:1: D101  Missing docstring in public class
claimstore/modules/claims/models.py:86:1: D102  Missing docstring in public method
claimstore/modules/claims/models.py:90:1: D101  Missing docstring in public class
claimstore/modules/claims/models.py:119:1: D102  Missing docstring in public method
claimstore/modules/claims/models.py:123:1: D101  Missing docstring in public class
claimstore/modules/claims/models.py:132:1: D102  Missing docstring in public method
claimstore/modules/claims/restful.py:1:1: D100  Missing docstring in public module
claimstore/modules/claims/restful.py:1:1: I101 Imported names are in the wrong order. Should be Blueprint, jsonify, request
claimstore/modules/claims/restful.py:2:1: I100 Imports statements are in the wrong order. from claimstore.modules.claims.models should be before from flask
claimstore/modules/claims/restful.py:2:1: I201 Missing newline before sections or imports.
claimstore/modules/claims/restful.py:2:80: E501 line too long (87 > 79 characters)
claimstore/modules/claims/restful.py:3:1: I100 Imports statements are in the wrong order. from claimstore.app should be before from claimstore.modules.claims.models
claimstore/modules/claims/restful.py:5:1: I100 Imports statements are in the wrong order. from claimstore.core.decorators should be before from claimstore.core.json
claimstore/modules/claims/restful.py:17:1: D103  Missing docstring in public function
claimstore/modules/claims/restful.py:34:80: E501 line too long (98 > 79 characters)
claimstore/modules/claims/restful.py:48:80: E501 line too long (98 > 79 characters)
claimstore/modules/claims/restful.py:53:1: D103  Missing docstring in public function
claimstore/modules/claims/restful.py:63:80: E501 line too long (86 > 79 characters)
claimstore/modules/claims/restful.py:65:80: E501 line too long (92 > 79 characters)
claimstore/modules/claims/restful.py:67:80: E501 line too long (90 > 79 characters)
claimstore/modules/claims/restful.py:69:80: E501 line too long (90 > 79 characters)
claimstore/modules/claims/restful.py:71:80: E501 line too long (89 > 79 characters)
claimstore/modules/claims/restful.py:74:80: E501 line too long (118 > 79 characters)
claimstore/modules/claims/restful.py:76:80: E501 line too long (87 > 79 characters)
claimstore/modules/claims/restful.py:78:80: E501 line too long (92 > 79 characters)
claimstore/modules/claims/restful.py:100:1: D103  Missing docstring in public function
```

--- a/claimstore/__init__.py
+++ b/claimstore/__init__.py
@@ -17,3 +17,5 @@
 # along with ClaimStore; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307,
 # USA.
+
+"""Main claimstore module."""

--- a/claimstore/app.py
+++ b/claimstore/app.py
@@ -18,6 +18,8 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307,
 # USA.
 
+"""Flask app creation."""
+
 from flask import Flask, render_template, jsonify
 from flask.ext.sqlalchemy import SQLAlchemy
 from claimstore.core.exception import InvalidUsage
@@ -28,12 +30,14 @@ db = SQLAlchemy()
 
 
 def handle_invalid_usage(error):
+    """Handle invalid usage exception."""
     response = jsonify(error.to_dict())
     response.status_code = error.status_code
     return response
 
 
 def create_app(db_create_all=False):
+    """Create flask app."""
     app = Flask(__name__)
 
     # Configurations

--- a/claimstore/core/__init__.py
+++ b/claimstore/core/__init__.py
@@ -17,3 +17,5 @@
 # along with ClaimStore; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307,
 # USA.
+
+"""Core module for ClaimStore."""

--- a/claimstore/core/decorators.py
+++ b/claimstore/core/decorators.py
@@ -18,11 +18,14 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307,
 # USA.
 
+"""Useful decorators."""
+
 from functools import wraps
 from flask import jsonify, request
 
 
 def only_json(f):
+    """Fail if the request header Content-Type is not application/json."""
     @wraps(f)
     def wrapper(*args, **kwds):
         if 'application/json' in request.headers['Content-Type']:

--- a/claimstore/core/exception.py
+++ b/claimstore/core/exception.py
@@ -18,11 +18,21 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307,
 # USA.
 
+"""Definition of specific exceptions."""
+
 
 class InvalidUsage(Exception):
+
+    """Invalid request.
+
+    Used to identify that a request is not correct (e.g. it does not follow the
+    proper JSON schema).
+    """
+
     status_code = 400
 
     def __init__(self, message, status_code=None, payload=None, details=None):
+        """Initialise the exception."""
         Exception.__init__(self)
         self.message = message
         if status_code is not None:
@@ -30,6 +40,7 @@ class InvalidUsage(Exception):
         self.payload = payload
 
     def to_dict(self):
+        """Return exception as a dictionary."""
         rv = dict(self.payload or ())
         rv['status'] = 'error'
         rv['message'] = self.message

--- a/claimstore/core/json.py
+++ b/claimstore/core/json.py
@@ -18,6 +18,8 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307,
 # USA.
 
+"""Useful JSON-related methods."""
+
 import os
 import json
 import jsonschema
@@ -27,10 +29,10 @@ from flask import current_app
 def get_json_schema(schema):
     """Return a given json schema.
 
-    The argument schema should have the format module.schema_name
-    (e.g. claims.claimants)
+    :param schema: schema to be fetched. It must be a string with the format:
+                   module.schema_name (e.g. claims.claimants).
+    :return: a str with the requested json schema.
     """
-
     module_name, schema_name = schema.split(".")
     schema_file_path = os.path.join(
         current_app.config['BASE_DIR'],
@@ -47,8 +49,13 @@ def get_json_schema(schema):
 
 
 def validate_json(json_input, schema):
-    """Validate JSON against a given schema."""
+    """Validate JSON against a given schema.
 
+    :param json_input: a dict with the full json to be validated.
+    :param schema: JSON schema to use in the validation. It must be a string
+                   with the format module.schema_name (e.g. claims.claimants).
+    :return: True if json_input follows the schema. False otherwise.
+    """
     if schema:
         schema_content = get_json_schema(schema)
         jsonschema.validate(json_input, json.loads(schema_content))

--- a/claimstore/modules/__init__.py
+++ b/claimstore/modules/__init__.py
@@ -17,3 +17,5 @@
 # along with ClaimStore; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307,
 # USA.
+
+"""Main package for modules."""

--- a/claimstore/modules/claims/__init__.py
+++ b/claimstore/modules/claims/__init__.py
@@ -17,3 +17,5 @@
 # along with ClaimStore; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307,
 # USA.
+
+"""Claims module."""

--- a/claimstore/modules/claims/models.py
+++ b/claimstore/modules/claims/models.py
@@ -18,12 +18,17 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307,
 # USA.
 
+"""ClaimStore data model."""
+
 from uuid import uuid4
 from sqlalchemy.dialects.postgresql import UUID, JSONB
 from claimstore.app import db
 
 
 class Claim(db.Model):
+
+    """Represents a Claim."""
+
     uid = db.Column(
         db.Integer,
         primary_key=True
@@ -77,10 +82,14 @@ class Claim(db.Model):
     claim_details = db.Column(JSONB)
 
     def __repr__(self):
+        """Printable version of the Claim object."""
         return '<Claim {}>'.format(self.uuid)
 
 
 class Claimant(db.Model):
+
+    """Represents a Claimant."""
+
     uid = db.Column(
         db.Integer,
         primary_key=True,
@@ -104,10 +113,14 @@ class Claimant(db.Model):
     url = db.Column(db.String)
 
     def __repr__(self):
+        """Printable version of the Claimant object."""
         return '<Claimant {}>'.format(self.uuid)
 
 
 class IdentifierType(db.Model):
+
+    """Represents an identifier type."""
+
     uid = db.Column(db.Integer, primary_key=True)
     name = db.Column(
         db.String,
@@ -137,10 +150,18 @@ class IdentifierType(db.Model):
     )
 
     def __repr__(self):
+        """Printable version of the IdentifierType object."""
         return '<IdentifierType {}>'.format(self.name)
 
 
 class Predicate(db.Model):
+
+    """Represents a predicate.
+
+    The predicate defines the type of claim. An example of predicate could be:
+    is_same_as.
+    """
+
     uid = db.Column(db.Integer, primary_key=True)
     name = db.Column(
         db.String,
@@ -150,4 +171,5 @@ class Predicate(db.Model):
     )
 
     def __repr__(self):
+        """Printable version of the Predicate object."""
         return '<Predicate {}'.format(self.name)

--- a/claimstore/modules/claims/restful.py
+++ b/claimstore/modules/claims/restful.py
@@ -18,6 +18,8 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307,
 # USA.
 
+"""Restful resources for the claims module."""
+
 from flask import Blueprint, request, jsonify
 from claimstore.modules.claims.models import Claim, Claimant, IdentifierType, \
     Predicate
@@ -36,6 +38,11 @@ claims_restful = Blueprint(
 @claims_restful.route('/subscribe', methods=['POST'])
 @only_json
 def subscribe():
+    """Register a new claimant in the database.
+
+    This POST service expects JSON data following the JSON schema defined for
+    claimants.
+    """
     json_data = request.get_json()
 
     try:
@@ -75,6 +82,11 @@ def subscribe():
 @claims_restful.route('/claims', methods=['POST'])
 @only_json
 def submit_claim():
+    """Record a new claim.
+
+    This POST service expects JSON data following the JSON schema defined for
+    claims.
+    """
     json_data = request.get_json()
 
     try:
@@ -136,6 +148,7 @@ def submit_claim():
 
 @claims_restful.route('/claims', methods=['GET'])
 def get_claim():
+    """GET service that returns the stored claims."""
     return jsonify(
         json_list=[
             {'created': c.created,

--- a/claimstore/modules/claims/views.py
+++ b/claimstore/modules/claims/views.py
@@ -18,6 +18,8 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307,
 # USA.
 
+"""ClaimStore views."""
+
 from flask import render_template, Blueprint
 from claimstore.core.json import get_json_schema
 
@@ -30,11 +32,13 @@ claims_views = Blueprint(
 
 @claims_views.route('/', methods=['GET'])
 def index():
+    """Render the home page for ClaimStore."""
     return render_template('cover.html', active_menu='home')
 
 
 @claims_views.route('/subscription', methods=['GET'])
 def subscription():
+    """Render the subscription form page."""
     return render_template(
         "subscription.html",
         json_schema=get_json_schema('claims.claimant')
@@ -43,9 +47,11 @@ def subscription():
 
 @claims_views.route('/api', methods=['GET'])
 def api():
+    """Render the api documenation page."""
     return render_template("api.html", active_menu='api')
 
 
 @claims_views.route('/contact', methods=['GET'])
 def contact():
+    """Render the contact page."""
     return render_template("contact.html", active_menu='contact')

--- a/claimstore/wsgi.py
+++ b/claimstore/wsgi.py
@@ -18,6 +18,8 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307,
 # USA.
 
+"""WSGI app creation."""
+
 from __future__ import absolute_import
 from claimstore.app.app import create_app
 

--- a/run.py
+++ b/run.py
@@ -18,11 +18,14 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307,
 # USA.
 
+"""Basic run module."""
+
 import argparse
 from claimstore.app import create_app
 
 
 def main():
+    """Create app and run server."""
     parser = argparse.ArgumentParser()
     parser.add_argument(
         "--initdb",

